aggregate/utils/pageviews.py

```diff
@@ -238,32 +238,35 @@ def __init__(self):
         self.browser_commerce_steps = {}
         self.data = {}
 
-    def __process_commerce(self, commerce_file):
+    def __process_commerce(self, commerce_file, csv_delimiter):
         print("BrowserParser - processing commerce data from: " + commerce_file)
         with open(commerce_file) as csv_file:
-            r = csv.DictReader(csv_file, delimiter=';')
+            r = csv.DictReader(csv_file, delimiter=csv_delimiter)
             for row in r:
-                if row['browser_id']:
-                    if row['browser_id'] not in self.browser_commerce_steps:
-                        self.browser_commerce_steps[row['browser_id']] = {
-                            'checkout': 0,
-                            'payment': 0,
-                            'purchase': 0,
-                            'refund': 0
-                        }
-
-                    if row['step'] not in ['checkout', 'payment', 'purchase', 'refund']:
-                        raise Exception(
-                            "unknown commerce step: " + row['step'] + ' for browser_id: ' + row['browser_id'])
-                    else:
-                        self.browser_commerce_steps[row['browser_id']][row['step']] += 1
-
-    def __process_pageviews(self, f):
+                try:
+                    if row['browser_id']:
+                        if row['browser_id'] not in self.browser_commerce_steps:
+                            self.browser_commerce_steps[row['browser_id']] = {
+                                'checkout': 0,
+                                'payment': 0,
+                                'purchase': 0,
+                                'refund': 0
+                            }
+
+                        if row['step'] not in ['checkout', 'payment', 'purchase', 'refund']:
+                            raise Exception(
+                                "unknown commerce step: " + row['step'] + ' for browser_id: ' + row['browser_id'])
+                        else:
+                            self.browser_commerce_steps[row['browser_id']][row['step']] += 1
+                except KeyError:
```

Comment:
When can this KeyError happen? Why do we silently let it continue instead of failing?

Reply:
When the browser_id is empty in exported csv, which can happen during testing, then the aggregation thwors an exception and does not finish. Then it is not possible to test the whole process of aggregation. This is a workaround only for an empty browser_id. There should not be an empty browser_id in the production export. So this won't matter.

Reply:
There is now configurable option to ignore empty browser id only when appropriate cmd line argument is passed.